### PR TITLE
feat(cloudflare-client): export CLOUDFLARE_LOGPUSH_FIELDS (LLMO-5869)

### PR DESCRIPTION
**Pull request by IrisAlexandrescu**
## Summary

- Repo 1 of the LLMO-5869 Cloudflare Logpush automation 3-repo plan.
- Cloudflare Logpush job methods (`requestLogpushOwnership`, `listLogpushJobs`, `createLogpushJob`) were already added to `CloudflareClient` in a prior commit on this branch.
- This commit adds `CLOUDFLARE_LOGPUSH_FIELDS`, exported from `cdn-helpers.js`, as the single source of truth for the 9 `byocdn-cloudflare` Logpush field names — needed by the `spacecat-api-service` controller (separate PR) to build a real Logpush job's `output_options.field_names` without duplicating the literal array. `prettifyLogForwardingConfig`'s own `byocdn-cloudflare` formatter now references this same constant instead of an inline duplicate, so there is exactly one place this list lives.

## Related Issues

LLMO-5869

## Test plan

- [x] `packages/spacecat-shared-utils` — 46 tests passing, 100% statements/lines, 96.55% branch (pre-existing gap, unrelated to this change)
- [x] `npx eslint` clean on changed files

## Downstream

`spacecat-api-service`'s Cloudflare Logpush controller (LLMO-5869, separate draft PR) depends on this being merged + published (semantic-release) before its dependency bump on `@adobe/spacecat-shared-utils` and `@adobe/spacecat-shared-cloudflare-client` can land for real.

### CODE REVIEW

**Review by MysticatBot — APPROVED** [review]
Hey @IrisAlexandrescu,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - clean, well-scoped addition following established patterns.
**Complexity: MEDIUM** - medium diff across two packages.
_Changes: Adds three Cloudflare Logpush API methods to CloudflareClient and extracts a shared CLOUDFLARE_LOGPUSH_FIELDS constant into spacecat-shared-utils (7 files)._

<details>
<summary>Non-blocking (1): minor issues and suggestions</summary>

- nit: `CLOUDFLARE_LOGPUSH_FIELDS` is a plain mutable array; any consumer can accidentally `.push()` and corrupt the shared reference - `packages/spacecat-shared-utils/src/cdn-helpers.js:49`. Consider `export const CLOUDFLARE_LOGPUSH_FIELDS = Object.freeze([...])` to communicate immutability intent at zero runtime cost.

</details>

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 39s | <b>Cost:</b> $5.01 | <b>Commit:</b> <code>68cfb5bc81ad317f12982e60c38f93187f58198c</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by nit23uec — APPROVED** [review]

File: packages/spacecat-shared-cloudflare-client/src/index.d.ts [hunk 1]
```diff
@@ -82,4 +118,13 @@ export default class CloudflareClient {
   addRoute(zoneId: string, pattern: string, scriptName: string): Promise<WorkerRoute>;
 
   deleteRoute(zoneId: string, routeId: string): Promise<object>;
+
+  requestLogpushOwnership(
```
Comment (MysticatBot) [inline packages/spacecat-shared-cloudflare-client/src/index.d.ts:125]: issue (blocking): `getZone(zoneId)` is implemented in `cloudflare-client.js` and fully tested, but has no corresponding TypeScript declaration here. TypeScript consumers importing the client will get a compile error when calling `client.getZone(...)`.

Add before `requestLogpushOwnership`:

```typescript
getZone(zoneId: string): Promise<{id: string, name: string, status: string}>;
```

**Review by MysticatBot — CHANGES_REQUESTED** [review]
Hey @IrisAlexandrescu,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Request changes** - one type declaration gap blocks TypeScript consumers.
**Complexity: MEDIUM** - medium diff across two packages.
_Changes: Adds Cloudflare Logpush API methods and a getZone method to CloudflareClient, extracts a shared CLOUDFLARE_LOGPUSH_FIELDS constant (7 files)._

## Must fix before merge
1. [Important] Missing `getZone` TypeScript declaration - `packages/spacecat-shared-cloudflare-client/src/index.d.ts:122` (details inline)

<details>
<summary>Non-blocking (1): minor issues and suggestions</summary>

- nit: `CLOUDFLARE_LOGPUSH_FIELDS` test re-declares the exact array and asserts equality (tautological test) - `packages/spacecat-shared-utils/test/cdn-helpers.test.js:706`. Consider asserting structural properties (length, key members, no duplicates) instead, or add a comment marking it as an intentional contract snapshot.

</details>

## Previously flagged, now resolved
- Object.freeze on CLOUDFLARE_LOGPUSH_FIELDS now applied.

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 42s | <b>Cost:</b> $4.87 | <b>Commit:</b> <code>b5448ab21a0b24b09572df1cd3a227472c8c93d1</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

File: packages/spacecat-shared-cloudflare-client/src/index.d.ts [hunk 2]
```diff
@@ -77,9 +113,20 @@ export default class CloudflareClient {
 
   listZones(options?: ZoneListOptions): Promise<CloudflareZone[]>;
 
+  getZone(zoneId: string): Promise<CloudflareZone>;
```
Comment (MysticatBot) [inline packages/spacecat-shared-cloudflare-client/src/index.d.ts:117]: issue (blocking): `getZone` returns `Promise<CloudflareZone>` but `CloudflareZone` (line 31) only has `{id: string, name: string}`. The JSDoc documents `{id, name, status}` and the test asserts `status: 'active'`. TypeScript consumers cannot access `.status` without a type error.

Fix: add `status: string` to the `CloudflareZone` interface:

```typescript
export interface CloudflareZone {
  id: string;
  name: string;
  status: string;
}
```

File: packages/spacecat-shared-utils/src/index.js [hunk 3]
```diff
@@ -130,7 +130,7 @@ export {
   getSpacecatBotIps,
   formatAllowlistMessage,
 } from './bot-blocker-detect/bot-blocker-detect.js';
-export { prettifyLogForwardingConfig } from './cdn-helpers.js';
+export { prettifyLogForwardingConfig, CLOUDFLARE_LOGPUSH_FIELDS } from './cdn-helpers.js';
```
Comment (MysticatBot) [inline packages/spacecat-shared-utils/src/index.js:133]: issue (blocking): `CLOUDFLARE_LOGPUSH_FIELDS` is now exported here but has no corresponding entry in `src/index.d.ts`. TypeScript consumers importing it will get a compile error.

Fix: add to `src/index.d.ts` after `prettifyLogForwardingConfig`:

```typescript
export const CLOUDFLARE_LOGPUSH_FIELDS: readonly string[];
```

**Review by MysticatBot — CHANGES_REQUESTED** [review]
Hey @IrisAlexandrescu,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Request changes** - two TypeScript declaration gaps will break downstream TS consumers.
**Complexity: MEDIUM** - medium diff across two packages.
_Changes: Adds Cloudflare Logpush API methods and getZone to CloudflareClient, extracts a shared CLOUDFLARE_LOGPUSH_FIELDS constant (7 files)._

## Must fix before merge
1. [Important] `CloudflareZone` interface missing `status` field - `packages/spacecat-shared-cloudflare-client/src/index.d.ts:31`. The JSDoc on `getZone` documents the return as `{id, name, status}` and the test asserts `{ id, name: "example.com", status: "active" }`, but `CloudflareZone` only has `{id: string, name: string}`. TS consumers calling `getZone` cannot access `.status` without a type error. Fix: add `status: string` to the `CloudflareZone` interface.
2. [Important] Missing TypeScript declaration for `CLOUDFLARE_LOGPUSH_FIELDS` - `packages/spacecat-shared-utils/src/index.d.ts`. The constant is exported from `src/index.js` but has no corresponding entry in the `.d.ts` file. TS consumers importing it will get a compile error. Fix: add `export const CLOUDFLARE_LOGPUSH_FIELDS: readonly string[];` after the `prettifyLogForwardingConfig` declaration.

<details>
<summary>Non-blocking (1): minor issues and suggestions</summary>

- nit: `CLOUDFLARE_LOGPUSH_FIELDS` test re-declares the exact array and asserts equality (tautological test) - `packages/spacecat-shared-utils/test/cdn-helpers.test.js:706`. Consider asserting structural properties (length, all strings, no duplicates) or marking it as an intentional contract snapshot with a comment explaining why.

</details>

## Previously flagged, now resolved
- Object.freeze on CLOUDFLARE_LOGPUSH_FIELDS now applied.
- getZone TypeScript declaration added in index.d.ts.

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 4m 14s | <b>Cost:</b> $6.64 | <b>Commit:</b> <code>a6726e805ceb6f17f31c11dfdcb4eddf1fbe5994</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by MysticatBot — APPROVED** [review]
Hey @IrisAlexandrescu,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - clean, well-scoped addition following established patterns; prior findings resolved.
**Complexity: MEDIUM** - medium diff across two packages.
_Changes: Adds Cloudflare Logpush API methods and getZone to CloudflareClient, extracts a shared CLOUDFLARE_LOGPUSH_FIELDS constant (8 files)._

<details>
<summary>Non-blocking (4): minor issues and suggestions</summary>

- nit: `zoneId` and `dataset` are interpolated directly into URL paths without `encodeURIComponent` - a pre-existing pattern in this client, but worth hardening in a follow-up since this is a shared library. Defense-in-depth only; CF validates server-side. - `packages/spacecat-shared-cloudflare-client/src/cloudflare-client.js:232`
- suggestion: `createLogpushJob` serializes the entire `payload` object verbatim. Consider destructuring only the documented fields (`dataset`, `destination_conf`, `ownership_challenge`, `name`, `output_options`, `enabled`) before `JSON.stringify` to make the contract explicit and prevent accidental payload pollution. - `packages/spacecat-shared-cloudflare-client/src/cloudflare-client.js:114`
- nit: `CLOUDFLARE_LOGPUSH_FIELDS` test re-declares the exact array and asserts equality (tautological test). Consider asserting structural properties (length, all strings, no duplicates, frozen) to make the test independent of the implementation. - `packages/spacecat-shared-utils/test/cdn-helpers.test.js:491`
- suggestion: Add a test case for `createLogpushJob(zoneId, null)` to exercise the `!payload` guard explicitly - the code handles it correctly but the test suite does not prove it. - `packages/spacecat-shared-cloudflare-client/test/cloudflare-client.test.js`

</details>

## Previously flagged, now resolved
- CloudflareZone interface now includes `status: string`.
- `CLOUDFLARE_LOGPUSH_FIELDS` TypeScript declaration added to `src/index.d.ts`.
- `getZone` TypeScript declaration added.

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 40s | <b>Cost:</b> $3.59 | <b>Commit:</b> <code>ec01ebe2363fc18628d8810e4dd2c095259c01f9</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>
